app/assets/javascripts/discourse/app/components/post/menu/buttons/like.gjs

```diff
@@ -82,17 +92,112 @@ export default class PostMenuLikeButton extends Component {
       </div>
     {{else}}
       <div class="double-button">
-        <LikeCount
-          ...attributes
-          @action={{@buttonActions.toggleWhoLiked}}
-          @state={{@state}}
-          @post={{@post}}
-        />
+        {{#if @post.likeCount}}
+          <LikedUsersList ...attributes @post={{@post}} />
+        {{else}}
+          <LikeCount
+            ...attributes
+            @action={{@buttonActions.toggleWhoLiked}}
+            @state={{@state}}
+            @post={{@post}}
+          />
+        {{/if}}
       </div>
     {{/if}}
   </template>
 }
 
+class LikedUsersList extends Component {
```

Comment:
I'd extract all those components into their own files, in `.../components/post/menu/buttons/like/`